### DISCUSSION TITLE
Why "SumatraPDF is running as admin and cannot open files from a non-admin process" error?

### DISCUSSION THREAD

**Discussion by kjk** [sumatrapdfreader/sumatrapdf#2316] [Documentation]
# Things changed in 3.7

In 3.7 instead of showing error message it'll just work by starting a new non-admin process, ignoring (in this specific scenario) `ReuseInstance` setting

# OBSOLETE INFORMATION AS OF 3.7

When you try to open e.g. a PDF file by double-clicking you might see an error message `SumatraPDF is running as admin and cannot open files from a non-admin process`.

**What does it mean?**

Unfortunately this is somewhat complicated and you need to understand several things about Windows and SumatraPDF and how they interact.

On Windows a process can run with elevated administrator privileges. This applies to SumatraPDF as well.

SumatraPDF implements a single process mode, When you try to open a file with `SumatraPDF.exe <path to a file>` and SumatraPDF is already running, we don't start a new process but send path to a file to existing process via Windows mechanism called DDE.

For security reasons Windows doesn't allow sending DDE messages from non-admin process to admin process.

When you double-click on a file in e.g. file explorer, what happens under the covers is that Windows simply calls `SumatraPDF.exe <path to a file>`.

So when you see this message, likely sequence of events was:
* SumatraPDF was started in admin mode
* you double-click on a file which runs `SumatraPDF.exe <path to a file>` in non-admin mode. This non-admin SumatraPDF process needs to send a file path to existing SumatraPDF process via DDE but we can't due to security restrictions implemented by Windows

But how did SumatraPDF started in admin mode in the first place?

One case is running Everything (a file search app) in admin mode and opening a file in SumatraPDF from Everything.

There are many other ways to start a process in admin mode ( use `Run as administrator` context menu, run from a terminal app running in Admin mode).

**How to solve it?**

Unfortunately the only option is to close the admin SumatraPDF process.

If this is due to Everything, you can re-configure it to run in non-admin mode as described at https://www.voidtools.com/faq/#how_do_i_prevent_the_uac_prompt_when_running_everything

**Comment by Ander3232** [comment]
> If this is due to Everything, you can re-configure it to run in non-admin mode.

Will you please tell me how to do that? I have this problem only when trying to use Sumatra to open PDFs from a Web browser. I checked my Sumatra .exe file's Properties, but there's no "Run as Administrator" option to un-check there. (The shortcut I use to run Sumatra manually does have that option, but it's already un-checked.) Is this something that can be done in the Registry? Thanks!

**Reply by kjk** [reply]
https://www.voidtools.com/faq/#how_do_i_prevent_the_uac_prompt_when_running_everything

**Reply by Ander3232** [reply]
If I'm following you, you're using the example of trying to open PDFs from a file-search utility called "Everything" (not referring to running "every" process some way, right?). And the page you've pointed me to—if I'm reading _that_ right!—says the problem can be solved by running Everything as a normal user, not an administrator.

But I'm already running my browser as a normal user, not an admin. I log in as an admin, but the browser shortcut I use has the "Run as administrator" checkbox un-checked. My understanding is that, even when you're logged in as an admin, you must give programs specific permission to run in admin mode (but only admins can do that, right?). Or am I confused about that too...?

**Reply by Ander3232** [reply]
Oh, sorry, I thought I'd gotten back here about this. Too many distractions...

Apparently a Windows Update set Microsoft Edge as my default PDF app. I tried setting it back to Sumatra, but it had no effect until I restarted Windows.

I still have no idea what your suggestion meant, but I appreciate your time anyway. Cheers!

**Comment by Ander3232** [comment]
The funny thing is, when this error occurs in Sumatra, it shows a link that says "Learn more about this error", which points to this page. So why the vague reference to the "Everything" app, etc.?

Also, while I do use an Administrator account in Windows 10, I don't have Sumatra's exe set to "Run as an administrator". So why does trying to open a subsequent document in it cause a problem? Am I suddenly less of an "administrator" than I was when I opened the first one? 

Somehow, Acrobat and Foxit have found a way around this limitation (or didn't have it in the first place). Sumatra's much lighter and simpler, so I'd rather use it—but I'm not sure it's worth the trouble if this can't be improved. If it's really a security precaution of some kind, couldn't there at least be an option to deactivate it? Thanks!

**Comment by GitHubRulesOK** [comment]
@Ander3232 
Not an expert on how to tell what the trigger is, but reading between the lines 
"Close the admin SumatraPDF process" is the solution, but you are in catch22 since the cause of elevation, may well be the call of a downloaded web file in an elevated browser, MS have likely raised your status when asking to open a copy of remote pdf. Try to download a pdf, then in downloads folder right click and set the PDF properties to "unblock". Then outside of the secured browser start the pdf session anew with the "trusted" pdf. If that then avoids the error message, you at least know the cause if not the cure.
![image](https://user-images.githubusercontent.com/12754913/234823743-5940a113-14fc-4a3a-ae6f-bde93877e013.png)

**Reply by Ander3232** [reply]
> @Ander3232 Not an expert on how to tell what the trigger is, but reading between the lines "Close the admin SumatraPDF process" is the solution, but you are in catch22 since the cause of elevation, may well be the call of a downloaded web file in an elevated browser, MS have likely raised your status when asking to open a copy of remote pdf. 

It doesn't happen just when I try to view PDFs from a browser... As I just explained, it also happens when I try to open a 2nd PDF via a shortcut on my own system. That's just goofy.

> Try to download a pdf, then in downloads folder right click and set the PDF properties to "unblock"...

I don't see any way to do that, and I've never even heard of such an option.

> ...Then outside of the secured browser start the pdf session anew with the "trusted" pdf. If that then avoids the error message, you at least know the cause if not the cure.

Sorry, you've totally lost me (and I'm a technical writer!). Thanks for trying to help—but as I've explained, no other PDF reader I've ever used has been this difficult. It's too bad, because I like everything else about Sumatra (its speed, simple GUI, how it doesn't continually try to sell you something). But this glitch doesn't make sense to me. If it is indeed some sort of security concern, we should be able to turn it off by choice.

Anyway, no more time to spend on this... It's back to Foxit, I guess 👋🏽

**Comment by wellerz** [comment]
I met this issue and the scenario is exactly what's mentioned by the author here.
Weird that no one mentions the simple workaround, which is to open a 2nd PDF file by finding it in "Everything" and double click.
Btw this PDF reader is really cool.

**Reply by Ander3232** [reply]
**wellerz > I met this issue and the scenario is exactly what's mentioned by the author here. Weird that no one mentions the simple workaround, which is to open a 2nd PDF file by finding it in "Everything" and double click...**

Okay, I didn't go back to Foxit. 😏 But I have no idea what you're talking about here—"Open a 2nd PDF file by finding it in 'Everything'" ? What's "Everything"? 

I'm afraid that makes no more sense to me than the other advice here, "Try to download a pdf, then in downloads folder right click and set the PDF properties to 'unblock'..." Is that a property in Linux, perhaps? It doesn't appear in PDF file properties in Windows.

**Reply by tactilis** [reply]
> What's "Everything"?

It's an application that can do very fast search for files based on name, partial name, file type, and many other attributes.

Highly recommended.

Check it out:  https://www.voidtools.com/faq/

**Reply by Ander3232** [reply]
Okay, but what does that have to do with Sumatra?

**Reply by kjk** [reply]
It's explained at the top:
> One case is running Everything (a file search app) in admin mode and opening a file in SumatraPDF from Everything.

**Reply by Ander3232** [reply]
**kjk > It's explained at the top...**

Okay, I see it:

**kjk> "But how did SumatraPDF started in admin mode in the first place? One case is running Everything (a file search app) in admin mode and opening a file in SumatraPDF from Everything."**

In other words, if you launch Sumatra from a process you started as an admin, then you try to open a another PDF from a non-admin process, Sumatra says, "Hold on! It's one thing if you want to cavalierly open a PDF in an environment where it's allowed to do practically anything... But you're asking me to let you open _another_ PDF the same way? On your own PC? What are you, crazy?"

Adding to the confusion, I'm _already_ an admin. My Windows account is an administrator account. Here, I'll show you:

![account_window](https://user-images.githubusercontent.com/25757204/235011023-1344b425-caf0-4463-80cf-84d988d29848.jpg)

Already being an Administrator, I never understood the need to set individual apps to "run as Administrator". (Wouldn't it make more sense to set them to _not_ run as an Administrator?) Is it possible that all an Administrator account means is that you can _authorize_ processes to run as what you already are? Gosh, wasn't life complicated enough?

Anyway, if I'm following you, what you're doing is pointing out that Sumatra is _more security-conscious_ than most PDF viewers. But I already knew that. If what I've read online is true, it _doesn't run scripts_ (e.g.: https://forum.sumatrapdfreader.org/t/how-safe-is-sumatra-can-it-execute-any-kind-of-code-or-script/976). Indeed, I've looked through all its settings, and JavaScript isn't even mentioned—it's not an issue. Sumatra is simple, fast and safe.

Of course, one could always click an embedded URL pointing to a shady site. But if people don't realize the possible consequences of clicking unfamiliar URLs, how will you stop them from doing that on webpages, or in email, or anywhere else that shady URLs can pop up these days? Will Sumatra monitor all that too? Will it be on hand to dissuade us from crossing busy streets on red lights, or from marrying that person we met in a bar last week? 😬💦

Okay, I've figured it out: All I have to do is set Sumatra itself to "run as Administrator". Now it doesn't balk if I want to view PDFs I find online. Which I'm doing voluntarily. To see them, on my own PC, using a _safe_, non-script-running PDF viewer. If someone had mentioned that earlier, I wouldn't have taken so much of your time.

Sometimes it just takes a while to figure these things out—sort of like everything you do in Linux. 😆 Cheers, A.

**Comment by mrjohnr** [comment]
for me seems solved reinstalling it as normal user,NOT  'Run as admin'

**Comment by SanderBouwhuis** [comment]
Can't you set a mutex or a registry entry? I use that myself to allow a new instance of a process to indirectly communicate with an existing instance of the same process.

**Reply by GitHubRulesOK** [reply]
unclear what you mean SumatraPDF works via Direct Dynamic Exchange DDE but Microsoft dont encourage tainting between 32bit and 64bit channels likewise a Lower UAC level cannot control a Higher UAC level Personally have no problems but then I stick to using same levels across the board (rarely need extra elevation except once or twice a week)

**Comment by SanderBouwhuis** [comment]
What I mean is that you can set a key in HKEY_CURRENT_USER which applications of all levels can access. It's not optimal, but better than an error message box.

**Comment by AnxiousPleaseHelp** [comment]
Might using SumatraPDF Portable solve this problem?
https://www.sumatrapdfreader.org/download-free-pdf-viewer

Edit: Portable

**Reply by kjk** [reply]
No, this behavior does not depend on portable vs. installed.

**Comment by StreamThreader** [comment]
I have this same error.
Then i opened Sumatra separated, and close all tabs, then sumatra can opened pdf file normaly.
I think some file in previous tab can not be open, and Sumatra stop working, until we close not accesible file.

**Comment by Zimbo3003** [comment]
I have the same error.
I open an attachment in an Outlook 365 email.
Both SumatraPDF and Outlook are running under the same user account, which is part of the Administrators group.

**Reply by GitHubRulesOK** [reply]
@Zimbo3003  ensure that whatever variant of SumatraPDF is that you run the SAME BITNESS so if SumatraPDF is 64bit then any other app you want it applied to must also be 64bit.
Or to put it another way.
 I am often trying to use SumatraPDF for a task it does not officially support like web browsing I must use 32bit SumatraPDF with a 32bit browser OR 64bit SumatraPDF with the 64bit Browser
